packages/chai-a11y-axe/src/accessible.js

```diff
@@ -0,0 +1,120 @@
+/* global axe */
+/**
+ * Creates a map of rules to disable during the test.
+ * @param {Array<String>?} ignored List of rule names to ignore during the test.
+ * @return {Object|undefined} List of rules to be passed to test configuration or
+ * `undefined` when not needed.
+ */
+function getRules(ignored) {
+  if (!ignored || !ignored.length) {
+    return undefined;
+  }
+  const result = {};
+  ignored.forEach(rule => {
+    result[rule] = { enabled: false };
+  });
+  return result;
+}
+
+/**
+ * Performs the test using AXE core.
+ * @param {Element} element The element to be used to perform the test on.
+ * @param {Object} opts AXE configuration options.
+ * @return {Promise} Promise resolved to the test results object
+ */
+function runTestAsync(element, opts) {
+  return new Promise((resolve, reject) => {
+    // @ts-ignore
```

Comment:
can we upstream types or would that be too onerous?

Reply:
I never worked with types. Not sure how to start :(

Reply:
That's a reasonable answer. Let's consider opening an issue on aXe's repo.

Reply:
Axe has types in `axe-core/typings/axe-core/axe-core-tests.ts`. But still, not sure how to use it in the script.